src/System.Buffers/src/System/Buffers/ArrayPool.cs

```diff
@@ -9,24 +9,36 @@
 namespace System.Buffers
 {
     /// <summary>
-    /// This class provides resource pooling for arrays of any type. This is can increase performance
-    /// of certain applications where lots of arrays are created and destroyed in rapid succession.
-    /// This class is thread-safe.
+    /// Provides a resource pool that enables reusing instances of type <see cref="T:T[]"/>. 
     /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Renting and returning buffers with an <see cref="ArrayPool{T}"/> can increase performance
+    /// in situations where arrays are created and destroyed frequently, resulting in significant
+    /// memory pressure on the garbage collector.
+    /// </para>
+    /// <para>
+    /// This class is thread-safe.  All members may be used by multiple threads concurrently.
+    /// </para>
+    /// </remarks>
     public abstract class ArrayPool<T>
     {
         /// <summary>The default number of arrays that are available for rent.</summary>
         private const int DefaultNumberOfArraysPerBucket = 50;
         /// <summary>The default length of each Rent'able array; equal to 1MB in bytes</summary>
```

Comment:
This should be more like `the maximum length of an array that will be eligible for being returned to the pool`. Keeps causing confusion as people think all the arrays given out are 1MB.